### ISSUE TITLE
[Error] downloading mp3 with Python and Pytubefix

### ISSUE THREAD

**Issue by JuanBindez** [JuanBindez/pytubefix#283]
_______
**Describe the bug**
I recently saw someone on a forum complaining about this error when trying to download an mp3, the error is due to the "?" but I realized that the mp3 parameter had a bug that was naming the filename by itself when in fact it should leave it up to the user to choose the name or if the user does not provide a filename then the library should do the correct naming with the prefix mp3, in addition the parameter is not passing through the file_system for a download compatible with each file system

__________
**code that was used that resulted in the bug**

```python
from pytubefix import YouTube

# put your code here

```

__________
**Expected behavior**

__________
**Screenshots**
![image](https://github.com/user-attachments/assets/bd8af496-14e4-4aa1-a13b-4380a00b7559)

__________
**Desktop (please complete the following information):**
```python
>>> from pytubefix import info
>>> 
>>> print(info())
{'OS': 'linux', 'Python': '3.11.6 (main, Apr 10 2024, 17:26:07) [GCC 13.2.0]', 'Pytubefix': '8.0.0'}
>>> 


```

__________
**Additional context**

**Comment by jhanley-com** [comment]
Please do not post text as a screenshot. Copy and paste into your post.

What is the URL? No, I am not going to try and hand type it from your image.

The error message indicates that PyTubeFix cannot create a file. That means something is probably wrong with the file name or location.

The error futher indicates Errno 22, which means the argument is invalid - further indicating a problem with the file name

I don't think this is a bug with PyTubeFix.

**Comment by JuanBindez** [comment]
> Please do not post text as a screenshot. Copy and paste into your post.
> 
> What is the URL? No, I am not going to try and hand type it from your image.
> 
> The error message indicates that PyTubeFix cannot create a file. That means something is probably wrong with the file name or location.
> 
> The error futher indicates Errno 22, which means the argument is invalid - further indicating a problem with the file name
> 
> I don't think this is a bug with PyTubeFix.

You are correct, but in addition to the error (OSError: [Errno 22] Invalid argument: ) there was a bug that was fixed here

**Comment by JuanBindez** [comment]
the mp3 parameter was not treating the characters that cause problems in Windows and that is why it generated the error, in addition there was another problem in relation to the filename which was named "alone"
